source/pkg/reconciler/resources/receive_adapter.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	v1alpha1 "knative.dev/eventing-rabbitmq/source/pkg/apis/sources/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/api/apps/v1"
+	"knative.dev/eventing/pkg/utils"
+	"knative.dev/pkg/kmeta"
+	"strconv"
+)
+
+type ReceiveAdapterArgs struct {
+	Image   string
+	Source  *v1alpha1.RabbitmqSource
+	Labels  map[string]string
+	SinkURI string
+	MetricsConfig string
+	LoggingConfig string
+}
+
+func MakeReceiveAdapter(args *ReceiveAdapterArgs) *v1.Deployment {
+	replicas := int32(1)
+
+	env := []corev1.EnvVar{
+		{
+			Name:  "RABBITMQ_BROKERS",
+			Value: args.Source.Spec.Brokers,
+		},
+		{
+			Name:  "RABBITMQ_TOPIC",
+			Value: args.Source.Spec.Topic,
+		},
+		{
+			Name: "RABBITMQ_USER",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: args.Source.Spec.User.SecretKeyRef,
+			},
+		},
+		{
+			Name: "RABBITMQ_PASSWORD",
```

Comment:
it would be better if creds came from a mounted volume, see the comment earlier about bindings.